aipcli/dial.go

```diff
@@ -130,18 +103,39 @@ func withDefaultPort(target string, port int) string {
 	return target
 }
 
-func generateTraceID() (string, error) {
+func withForceTrace(cmd *cobra.Command) grpc.DialOption {
+	traceID := generateTraceID()
+	spanID := generateSpanID()
+	return grpc.WithUnaryInterceptor(func(
+		ctx context.Context,
+		method string,
+		req interface{},
+		reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) error {
+		// See https://cloud.google.com/trace/docs/setup#force-trace
+		const header = "x-cloud-trace-context"
+		value := fmt.Sprintf("%s/%d;o=1", traceID, spanID)
+		cmd.PrintErrln(">> trace ID:", traceID)
+		if IsVerbose(cmd) {
+			cmd.PrintErrln(">> span ID:", spanID)
+			cmd.PrintErrln(">> trace header:", header, "=", value)
+		}
+		ctx = metadata.AppendToOutgoingContext(ctx, header, value)
+		return invoker(ctx, method, req, reply, cc, opts...)
+	})
+}
+
+func generateTraceID() string {
 	var id [16]byte
-	if _, err := rand.Read(id[:]); err != nil {
-		return "", err
-	}
-	return hex.EncodeToString(id[:]), nil
+	_, _ = rand.Read(id[:]) // panics on error
```

Comment:
Where is it documented that this panics? It is strange that it would return an error if it panics and I am not seeing anywhere in the documentation that this is the case

Reply:
https://go-review.googlesource.com/c/go/+/602497/10/src/crypto/rand/rand.go

Reply:
Interesting but if I understand correctly this is Go tip (master) and not the current Go release so maybe make that change then?